### PR TITLE
Add .github/workflows/checkstyle.yml

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -17,9 +14,7 @@ jobs:
 
       - uses: actions/setup-java@v4
         with:
-          distribution: corretto
-          java-version: 17
+          java-version-file: .tool-versions
```
Comment (motlin) [inline .github/workflows/build-and-test.yml:17]: @leifwalsh I just learned that actions/setup-java supports .tool-versions

File: .github/workflows/checkstyle.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: CheckStyle
+
+on:
+  pull_request:
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+    if: github.actor != 'dependabot[bot]'
```
Comment (motlin) [inline .github/workflows/checkstyle.yml:9]: I also just learned how to skip pointless checks like running CheckStyle on a Dependabot PR.

Comment (leifwalsh) [reply .github/workflows/checkstyle.yml:9]: What if it upgrades checkstyle?

Comment (motlin) [reply .github/workflows/checkstyle.yml:9]: Good point. I could undo this, or add the ability to trigger the workflow manually. WDYT?

Comment (leifwalsh) [reply .github/workflows/checkstyle.yml:9]: If it were me, I'd just remove the condition. It should almost never affect you on a dependabot PR anyway, it's not like it's slow/expensive

Comment (leifwalsh) [reply .github/workflows/checkstyle.yml:9]: (I should say also it's nice to know you can do this)

**Pull request merged**